groups/infrastructure/main.tf

```diff
@@ -0,0 +1,52 @@
+terraform {
+  required_version = "~> 1.3"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.54.0"
+    }
+    vault = {
+      source  = "hashicorp/vault"
+      version = "~> 3.18.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+
+terraform {
+  backend "s3" {}
+}
```

Comment:
The `backend` config should be a part of the main `terraform` config block at the top of the file.